apps/src/music/player/MusicPlayer.ts

```diff
@@ -25,61 +25,72 @@ const DEFAULT_KEY = Key.C;
  */
 export default class MusicPlayer {
   private readonly metricsReporter: LabMetricsReporter;
-  private bpm: number;
-  private key: Key;
-  private samplePlayer: SamplePlayer;
-  private library: MusicLibrary | null;
+  private readonly samplePlayer: SamplePlayer;
+
+  private bpm: number = DEFAULT_BPM;
+  private key: Key = DEFAULT_KEY;
 
   constructor(
     bpm: number = DEFAULT_BPM,
     key: Key = DEFAULT_KEY,
+    samplePlayer: SamplePlayer = new SamplePlayer(),
     metricsReporter: LabMetricsReporter = Lab2Registry.getInstance().getMetricsReporter()
   ) {
-    this.bpm = this.validateBpm(bpm);
-    this.key = this.validateKey(key);
-    this.samplePlayer = new SamplePlayer();
-    this.library = null;
+    this.samplePlayer = samplePlayer;
     this.metricsReporter = metricsReporter;
+    this.updateConfiguration(bpm, key);
   }
 
-  /**
-   * Initializes the MusicPlayer and {@link SamplePlayer} with the music library.
-   * Playback cannot start until the player is initialized.
-   */
-  initialize(
-    library: MusicLibrary,
-    updateLoadProgress: (value: number) => void
-  ) {
-    this.library = library;
-
-    // Set key and BPM from library if present
-    const libraryBpm = library.getBPM();
-    if (libraryBpm) {
-      this.bpm = this.validateBpm(libraryBpm);
+  updateConfiguration(bpm?: number, key?: Key) {
+    if (bpm) {
+      this.bpm = this.validateBpm(bpm);
+      this.samplePlayer.setBpm(this.bpm);
     }
-    const libraryKey = library.getKey();
-    if (libraryKey) {
-      this.key = this.validateKey(libraryKey);
+    if (key) {
+      this.key = this.validateKey(key);
     }
+  }
 
-    this.samplePlayer.initialize(library, this.bpm, updateLoadProgress);
+  setUpdateLoadProgress(updateLoadProgress: (value: number) => void) {
+    this.samplePlayer.setUpdateLoadProgress(updateLoadProgress);
   }
 
   /**
    * Load a sound into the sound system using the given buffer. Currently only
    * used to upload custom sounds (hidden/demo-only feature).
    *
+   * TODO: re-enable
```

Comment:
We can look into re-enabling or removing this as a follow-up. Worth noting that as far as I can tell, this isn't currently used (we may have removed the code to upload sounds when we reworked some of the UI)